### PR TITLE
Rename pagination to article-pagination and add new pagination style

### LINKED ISSUE #1975: Build page pagination component
A pattern in Vanilla needs to be flexible enough that it can be used across a wide range of scenarios and shouldn't be constrained to a specific number of applications.

## Visual
page-pagination-009

## Context
- We currently have pagination in Vanilla but it's only used for articles, we need to design a new variant of this pattern for pages
- Original design issue can be found here: https://github.com/ubuntudesign/vanilla-design/issues/169
- Page pagination design spec can be found here: https://github.com/ubuntudesign/vanilla-design/blob/master/Pagination/page-pagination.md

### PR DESCRIPTION
**Pull request by deadlight**
## Done

Built new pagination component and moved the old one

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/vanilla-framework/examples/patterns/pagination/
- Compare to the [design](https://github.com/ubuntudesign/vanilla-design/issues/169)

## Details

Fixes #1975

### CODE REVIEW

File: examples/patterns/pagination.html [hunk 1]
```diff
@@ -3,14 +3,83 @@
 title: Pagination
 category: _patterns
 ---
+<h4>Standard</h4>
```
Comment (anthonydillon) [inline examples/patterns/pagination.html]: Can we use the titles too, please? It's nicer to just have the mark up of the pattern to copy out.

File: examples/patterns/pagination.html [hunk 2]
```diff
@@ -3,14 +3,83 @@
...
+<ol class="p-pagination">
+  <li class="p-pagination__item p-pagination__item">
+    <a class="p-pagination__link p-pagination__link--previous" href="#previous" title="Previous page"><i class="p-icon--contextual-menu">Previous page</i></a>
```
[the remark this answers is not in the document]
Comment (deadlight) [reply examples/patterns/pagination.html]: @anthonydillon sure. I wasn't sure about the current guidance on classes. I'll clean that up.

**Comment by kwm14** [comment]
> @kwm14 The is an accessibility issue with the active links:
> 
> ```
> This element has insufficient contrast at this conformance level. 
> Expected a contrast ratio of at least 4.5:1, but text in this 
> element has a contrast ratio of 3.88:1. Recommendation: 
> change text colour to #006f97.
> ```
> Also, the spec only mentions it should inherit the `p-button--neutral` styling. The color of the button is text-color, not link color. I think the spec needs updating.

@anthonydillon yep the color of text color should be `$color-x-dark` we reverted the button styling update that recently go merged, due to the blue links not working across other components. Discussed here - https://github.com/vanilla-framework/vanilla-framework/pull/1994#issuecomment-423235636 in my design review I will add that the text-color needs updating 👍

**Comment by kwm14** [comment]
## Design review
@deadlight looking really good, tested in Chrome, Firefox and Safari. Just a couple of small design updates on top of what we discussed yesterday 😄 

- [ ] update text-color to `$color-x-dark` my fault for not letting you know about recent styling update
- [ ] number/chevron blocks should inherit `p-button--neutral` styling and is missing border-radius: 2px;

**Comment by kwm14** [comment]
LGTM @deadlight will review Percy now 👍

File: scss/_patterns_pagination.scss [hunk 3]
```diff
@@ -1,92 +1,74 @@
...
+      }
+
+      &:active {
+        background-color: darken($color-x-light, 15%);
+        border-color: $color-mid;
       }
 
+      &--previous,
       &--next {
         @extend .p-pagination__link; // sass-lint:disable-line placeholder-in-extend
```
Comment (anthonydillon) [inline scss/_patterns_pagination.scss]: Can you make the shared styles a placeholder please.

**Comment by deadlight** [comment]
@anthonydillon updated the branch

**Pull request merged**